### PR TITLE
Add blinder polynomials

### LINKED ISSUE #47: Add blinder polynomials
The Plonk protocol provides zero-knowledge by adding random multiples of the vanishing polynomial Z<sub>H</sub> (referred as blinder polynomials) to the witness based polynomials: The wire polynomials + permutation polynomial.

We should add these blinders.

### PR DESCRIPTION
**Pull request by davidnevadoc**
Closes #47

### CODE REVIEW

**Comment by bhgomes** [comment]
We should try to make these kinds of polynomial changes computable automatically. There are lots of magic numbers spread around like in this case the total degree, in this case, `7` and `8`. It will be hard to keep track of correctness over time as we keep adding new things, like lookups.

**Comment by ghost** [comment]
Just a side note, as Daniel Lubarov [describes](https://mirprotocol.org/blog/Adding-zero-knowledge-to-Plonk-Halo), when using the inner-product argument polynomial commitment scheme, this blinding should be done in a slightly different way for efficiency reasons. So while in theory it is independent of the commitment scheme it'll be tricky to abstract in the code.

**Comment by bhgomes** [comment]
Maybe it should be part of a trait design for polynomial commitment scheme then, in analogy with how commitment schemes have blinders which they use in the implementation in their own way.

**Review by mathcrypto — APPROVED** [review]

**Comment by mathcrypto** [comment]
I agree with @bhgomes, otherwise, it's confusing if we constantly change the polynomial degree. I also believe increasing the CRS degree should be justified (adding a side note of why this was needed)

**Review by mathcrypto — COMMENTED** [review]
Maybe open an issue about this so the temporary fix is replaced in the future

**Comment by davidnevadoc** [comment]
> Maybe open an issue about this so the temporary fix is replaced in the future

It is fixed in the next commit. Instead of getting the domain size form the `z_poly` degree it is passed directly to the function.

**Comment by mathcrypto** [comment]
Great then :)

File: src/constraint_system/helper.rs [hunk 1]
```diff
@@ -98,7 +103,7 @@ where
     // Compute Commit and Verifier Key
     let (sonic_ck, sonic_vk) = SonicKZG10::<E, DensePolynomial<E::Fr>>::trim(
         &universal_params,
-        verifier.circuit_size().next_power_of_two(),
+        verifier.circuit_size().next_power_of_two() + 7,
```
Comment (CPerezz) [inline src/constraint_system/helper.rs]: Is that a `+`? 
And if so, shouldn't it be 6?

**Review by CPerezz — COMMENTED** [review]
I would prefer to avoid the `+6` and add it internally.

And add in the docs of the functions that we add 6 to the user's expected DEGREE in order to fit in the blinders.

File: src/constraint_system/helper.rs (hunk 1 above)
Comment (davidnevadoc) [reply src/constraint_system/helper.rs]: You're right, should be a `+6`

**Comment by davidnevadoc** [comment]
There is an open discussion at [team-infra](https://discord.com/channels/918343325587746816/922469573670866954) about this topic.

Due to the negative impact in compiled circuit size and proving time derived from the modifications made in PR it will not be merged unless we have concluded that there is no better alternative.

File: src/constraint_system/helper.rs (hunk 1 above)
Comment (davidnevadoc) [reply src/constraint_system/helper.rs]: fixed

**Review by bhgomes — APPROVED** [review]

**Pull request merged**